ecma402_traits/src/collator.rs

```diff
@@ -0,0 +1,144 @@
+        Search,
+    }
+
+    /// Set the sensitivity of the collation.
+    ///
+    /// Which differences in the strings should lead to non-zero result values.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Sensitivity {
+        /// Only strings that differ in base letters compare as unequal.
+        /// Examples: a ≠ b, a = á, a = A.
+        Base,
+        /// Only strings that differ in base letters or accents and other diacritic marks compare
+        /// as unequal. Examples: a ≠ b, a ≠ á, a = A.
+        Accent,
+        /// Only strings that differ in base letters or case compare as unequal.
+        /// Examples: a ≠ b, a = á, a ≠ A.
+        Case,
+        /// Strings that differ in base letters, accents and other diacritic marks, or case compare
+        /// as unequal. Other differences may also be taken into consideration.
+        /// Examples: a ≠ b, a ≠ á, a ≠ A.
+        Variant,
+    }
+
+    /// Whether punctuation should be ignored.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Punctuation {
+        /// Ignore punctuation.
+        Ignore,
+        /// Honor punctyation.
+        Honor,
+    }
+
+    /// Whether numeric collation should be used, such that "1" < "2" < "10".
+    ///
+    /// This option can be set through an options property or through a Unicode extension key; if
+    /// both are provided, the options property takes precedence. Implementations are not required
+    /// to support this property.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Numeric {
+        /// Use numeric comparison.
+        Use,
+        /// Do not use numeric comparision.
+        Ignore,
+    }
+
+    /// Whether upper case or lower case should sort first.
+    ///
+    /// This option can be set through an options property or through a Unicode extension key; if
+    /// both are provided, the options property takes precedence. Implementations are not required
+    /// to support this property.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum CaseFirst {
+        /// Sort upper case first.
+        Upper,
+        /// Sort lower case first.
+        Lower,
+        /// Use locale default for case sorting.
+        False,
+    }
+}
+
+/// The options set by the user at construction time.  See discussion at the top level
+/// about the name choice.  Provides as a "bag of options" since we don't expect any
+/// implementations to be attached to this struct.
+///
+/// The default values of all the options are prescribed in by the [TC39 report][tc39lf].
+///
+/// [tc39lf] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Options {
+    pub usage: options::Usage,
+    pub sensitivity: options::Sensitivity,
+    pub punctuation: options::Punctuation,
+    pub numeric: options::Numeric,
+    pub case_first: options::CaseFirst,
+}
+
+impl Default for Options {
```

Comment:
If the spec indicates default values for all the enums, we should probably `impl Default` for them as well.

Reply:
I thought about this as well, and concluded that the default value of an enum at the site of use is not necessarily the "global" default.  From that perspective it didn't seem appropriate to prescribe a global default.